base/clistruct/clistruct.go

```diff
@@ -0,0 +1,152 @@
+// Package clistruct provides the ability to assign command line flags to fields in structs that declare `cli` and `help` struct tags.
+package clistruct
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"reflect"
+	"time"
+	"unsafe"
+)
+
+const (
+	cliStructNameTag = "cli"
+	cliStructHelpTag = "help"
+)
+
+// MustRegisterFlags will register flags on the given FlagSet from cli struct tags in the given struct.
+// FlagSet.Parse() must be called _after_ this has been run.
+//
+// Panics if the given struct cannot be assigned to a FlagSet (unsupported types, duplicate flags, etc.)
+func MustRegisterFlags(fs *flag.FlagSet, cliStruct interface{}) {
+	if err := registerFlags(fs, cliStruct); err != nil {
+		panic(err.Error())
```

Comment:
What's the benefit of panicking here instead of having MustRegisterFlags return error, and leave the error handling to the caller?

Reply:
Mainly for visibility (to us at dev time) - given we'll be registering this on startup with a struct, and it's known at startup time whether that's valid or not for a given build.

I'd be happy enough switching this to return `err` and ensure callers log as a fatal error.